### ISSUE TITLE
Adjust table sort states

### ISSUE THREAD

**Issue by cjcenizal** [vectara/vectara-ui#276]
We can improve the UX and accessibility in a few ways:
* Set `isSelected` on the column that's currently active for the sort. As you can see in the below screenshot you currently can't tell which one is active.
* Change the size of the icon button to be `s`. It's a bit too large next to he table header text, in its current state.
* Currently, clicking on one of the buttons toggles between "Sort ascending" and "Sort descending". But there's no way to return to the default sort order of the data, i.e. the sort order it was in when it was first retrieved from the API. Let's add a third state: "Remove sort". So as the user clicks on a button, it cycles from "Sort ascending" -> "Sort descending" -> "Remove sort" and back again. If a table is being sorted by column A, and then the user sorts by column B, the sort state of column A should be reset. So the first click on any column will always sort to "Sort ascending" on that column.
* The tooltips should be updated to reflect the outcome of clicking on that column's button. So they should be "Sort ascending" if clicking on it will set the sort state to be ascending, and so on. This will improve usability and accessibility.

<img width="1186" height="101" alt="Image" src="https://github.com/user-attachments/assets/38e20a15-e4ad-4acf-a6c6-2f820cb13479" />
